From: Ben Hutchings <benh@debian.org>
Date: Mon, 07 Aug 2023 19:24:36 +0200
Subject: x86/cpu: Avoid ABI change for GDS mitigations
Forwarded: not-needed

The changes to support control of GDS mitigation added some #includes
to arch/x86/kernel/cpu/common.c.  This changed the symbol version for
current_task which is defined there, not because of a real ABI change
but because it now includes definitions of more of the types referred
to by struct task_struct.  To avoid the symbol version change, make
those #includes conditional on !__GENKSYMS__.

---
--- a/arch/x86/kernel/cpu/common.c
+++ b/arch/x86/kernel/cpu/common.c
@@ -19,12 +19,16 @@
 #include <linux/kgdb.h>
 #include <linux/mem_encrypt.h>
 #include <linux/smp.h>
+#ifndef __GENKSYMS__
 #include <linux/cpu.h>
+#endif
 #include <linux/io.h>
 #include <linux/syscore_ops.h>
 
 #include <asm/stackprotector.h>
+#ifndef __GENKSYMS__
 #include <linux/utsname.h>
+#endif
 
 #include <asm/alternative.h>
 #include <asm/perf_event.h>
